alt_e2eshark/e2e_testing/backends.py

```diff
@@ -18,18 +18,23 @@
 class BackendBase(abc.ABC):
 
     @abc.abstractmethod
-    def compile(self, module: ModelArtifact) -> CompiledOutput:
+    def compile(self, module: ModelArtifact, extra_options : CompilerOptions) -> CompiledOutput:
         """specifies how to compile an MLIR Module"""
 
     @abc.abstractmethod
-    def load(self, artifact: CompiledOutput, func_name: str) -> Invoker:
+    def load(self, artifact: CompiledOutput, func_name: str, extra_options : RuntimeOptions) -> Invoker:
         """loads the function with name func_name from compiled artifact. This method should return a function callable from python."""
 
 
 from iree import compiler as ireec
 from iree import runtime as ireert
 
 
+def flag(arg : str) -> str:
+    if arg.startswith("--"):
+        return arg
+    else:
```

Comment:
I think you can leave out this else, and simply return.
```suggestion
    return f'--{arg}'
```

Nit: since arg is supposed to be an str, slicing could be a better option here given that arg is not too long in length.